spec/facility_spec.rb

```diff
@@ -23,4 +31,136 @@
+      @facility_1.registered_vehicles
+      expect(@facility_1.registered_vehicles). to eq []
+
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@cruz)
+      expect(@facility_1.registered_vehicles). to eq([@cruz])
+
+      @facility_2.register_vehicle(@bolt)
+      expect(@facility_2.registered_vehicles). to eq([])
+    end
+
+    it 'can collect fees' do
+      @facility_1.add_service('Vehicle Registration')
+      @facility_1.register_vehicle(@cruz)
+      expect(@facility_1.collected_fees). to eq(100)
+
+      @facility_2.register_vehicle(@bolt)
+      expect(@facility_2.collected_fees). to eq(0)
+
+    end
+  end
+
+  describe '#update registration' do
+    it 'can update registration date' do
+      @facility_1.register_vehicle(@cruz)
+      expect(@cruz.registration_date).to eq(Date.today)
+    end
+
+    it 'can update plate type' do 
+      @facility_1.register_vehicle(@camaro)
+      expect(@camaro.registration_date).to eq(Date.today)
+      expect(@camaro.calculate_plate_type). to eq(:antique)
+
+      @facility_1.register_vehicle(@bolt)
+      expect(@bolt.registration_date).to eq(Date.today)
+      expect(@bolt.calculate_plate_type). to eq(:ev)
+    end
+  end
+
+  describe '#Update services' do
+    it 'can administer written tests to registrants with permits' do
+
+      expect(@facility_1.administer_written_test(@registrant_1)).to eq(false)
+
+      @facility_1.add_service('Written Test')
+      expect(@facility_1.services).to eq(['Written Test'])
+      expect(@facility_1.administer_written_test(@registrant_1)).to eq(true)
+
+      expect(@facility_1.administer_written_test(@registrant_2)).to eq(false)
+      @registrant_2.earn_permit
+      expect(@facility_1.administer_written_test(@registrant_2)).to eq(true)
+    end
+  end
+
+  describe '#Update license data' do
+    it 'can update written' do
+
+      @facility_1.add_service('Written Test')
+      @facility_1.administer_written_test(@registrant_1)
+      expect(@registrant_1.license_data). to eq({:written=>true, :license=>false, :renewed=>false})
+    end
+  end
+
+  describe ' @Update services' do
+    it 'can administer road tests' do
+
+      @facility_1.add_service('Written Test')
+      @facility_1.add_service('Road Test')
+      expect(@facility_1.services).to eq(['Written Test', 'Road Test']) 
+      
+      @facility_1.administer_written_test(@registrant_1)
+      expect(@facility_1.administer_road_test(@registrant_1)).to eq(true)
+    end
+  end
+    
+  describe '#Update license data' do
+    it 'can update license' do
+
+        @facility_1.add_service('Written Test')
+        @facility_1.add_service('Road Test')
+        expect(@facility_1.services).to eq(['Written Test', 'Road Test']) 
+        
+        @facility_1.administer_written_test(@registrant_1)
+        @facility_1.administer_road_test(@registrant_1)
+        expect(@registrant_1.license_data). to eq({:written=>true, :license=>true, :renewed=>false})
```

Comment:
you could just check @registrant_1.license_data[:license] here, but this works too!